### PR TITLE
FCBHDBP-579 bibleis-web: invalid CDN request

### PR DESCRIPTION
**Pull request by vichugofsl**
# Description
The issue causing an infinite loop when a video chunk failed has been resolved. The fix involved eliminating a poor practice of retrying a failed request. Additionally, I've mitigated around 70 vulnerabilities by overriding three packages and removing some that are no longer necessary. Finally, I have overwritten the Load Policies: `playlistLoadPolicy`, `fragLoadPolicy` and `keyLoadPolicy` to use the exponential strategy for the backoff parameter.

However, I am still encountering some CORS errors for specific videos, such as /bible/FRAPDF/MAT/1.

This PR is related with the API DBP PR:
- https://github.com/faithcomesbyhearing/dbp/pull/932

## Related Issue

https://fullstacklabs.atlassian.net/browse/FCBHDBP-580

## How to Test
- You should go the path: `bible/EN1ESV/MAT/1` and it must load the correct page and the console doesn't render errors. Also, the CDN URL must be: `d1gd73roq7kqw6.cloudfront.net`
- You should go the path: `bible/FRAPDF/MAT/1` and it must load the correct page and the console doesn't render errors. Also, the CDN URL must be: `d1gd73roq7kqw6.cloudfront.net`

### CODE REVIEW

File: app/containers/VideoPlayer/index.js [hunk 1]
```diff
@@ -296,12 +297,76 @@ class VideoPlayer extends React.PureComponent {
...
+        default: {
+          maxTimeToFirstByteMs: 10000,
+          maxLoadTimeMs: 20000,
+          timeoutRetry: {
+            maxNumRetry: 2,
+            retryDelayMs: 0,
+            maxRetryDelayMs: 0,
+          },
+          errorRetry: {
+            maxNumRetry: 2,
```
Comment (bradflood) [inline app/containers/VideoPlayer/index.js:311]: I think we should retry at least four times for all situations; two is not enough.

**Review by bradflood — CHANGES_REQUESTED** [review]

File: app/containers/VideoPlayer/index.js (hunk 1 above)
Comment (vichugofsl) [reply app/containers/VideoPlayer/index.js]: @bradflood , I have updated it and all situations will have maxNumRetry value greater or equal than 4.

File: app/containers/VideoPlayer/index.js [hunk 2]
```diff
@@ -296,12 +297,76 @@ class VideoPlayer extends React.PureComponent {
...
     }
-    this.hls = new this.Hls();
+    this.hls = new this.Hls({
+      playlistLoadPolicy: {
+        default: {
+          maxTimeToFirstByteMs: 10000,
+          maxLoadTimeMs: 20000,
+          timeoutRetry: {
+            maxNumRetry: 4,
+            retryDelayMs: 0,
```
Comment (bradflood) [inline app/containers/VideoPlayer/index.js:307]: now the delay and maxDelay look wrong. how about retryDelayMs 1000 and maxRetryDelayMs 20000. Or are these specific to timeout retries, in which the delay has already been baked in?

Comment (vichugofsl) [reply app/containers/VideoPlayer/index.js]: @bradflood I have updated the parameters 'retryDelayMs' and 'maxRetryDelayMs' to 1000 and 20000 respectively, based on the documentation. This adjustment ensures that the system retries with the specified delay time after encountering a timeout error.

**Review by bradflood — APPROVED** [review]

**Pull request merged**